### PR TITLE
signet miner followups

### PR DESCRIPTION
**Pull request by ajtowns**
Followups from #19937

### CODE REVIEW

**Review by kallewoof — APPROVED** [review]
tested ACK f30c348be188b32a32f758c7e01b3bb0010e3125

Running this code on my side as of a minute ago.

**Comment by MarcoFalke** [comment]
The gitian builds are broken right now:

```

  CXXLD    bitcoin-util
bitcoin_util-bitcoin-util.o: In function `grind_task':
/home/ubuntu/build/bitcoin/distsrc-riscv64-linux-gnu/src/bitcoin-util.cpp:98: undefined reference to `__atomic_exchange_1'
collect2: error: ld returned 1 exit status
Makefile:4964: recipe for target 'bitcoin-util' failed
make[2]: *** [bitcoin-util] Error 1
make[2]: Leaving directory '/home/ubuntu/build/bitcoin/distsrc-riscv64-linux-gnu/src'
Makefile:14421: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/home/ubuntu/build/bitcoin/distsrc-riscv64-linux-gnu/src'
Makefile:809: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
```

```
  CXXLD    qt/bitcoin-qt.exe
make[2]: Leaving directory '/home/ubuntu/build/bitcoin/distsrc-x86_64-w64-mingw32/src'
make[1]: Leaving directory '/home/ubuntu/build/bitcoin/distsrc-x86_64-w64-mingw32/src'
Making all in doc/man
make[1]: Entering directory '/home/ubuntu/build/bitcoin/distsrc-x86_64-w64-mingw32/doc/man'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/ubuntu/build/bitcoin/distsrc-x86_64-w64-mingw32/doc/man'
make[1]: Entering directory '/home/ubuntu/build/bitcoin/distsrc-x86_64-w64-mingw32'
make[1]: Nothing to be done for 'all-am'.
make[1]: Leaving directory '/home/ubuntu/build/bitcoin/distsrc-x86_64-w64-mingw32'
+ make -j1 -C src check-security
make: Entering directory '/home/ubuntu/build/bitcoin/distsrc-x86_64-w64-mingw32/src'
Checking binary security...
bitcoin-util.exe: failed RELOC_SECTION
make: *** [check-security] Error 1
Makefile:14974: recipe for target 'check-security' failed
make: Leaving directory '/home/ubuntu/build/bitcoin/distsrc-x86_64-w64-mingw32/src'

**Comment by fanquake** [comment]
> bitcoin-util.exe: failed RELOC_SECTION

This will fix the reloc section issue (same as in #18702):
```diff
diff --git a/src/bitcoin-util.cpp b/src/bitcoin-util.cpp
index b702a68bd..30b809871 100644
--- a/src/bitcoin-util.cpp
+++ b/src/bitcoin-util.cpp
@@ -181,7 +181,16 @@ static int CommandLineUtil(int argc, char* argv[])
     return nRet;
 }
 
+#ifdef WIN32
+// Export main() and ensure working ASLR on Windows.
+// Exporting a symbol will prevent the linker from stripping
+// the .reloc section from the binary, which is a requirement
+// for ASLR. This is a temporary workaround until a fixed
+// version of binutils is used for releases.
+__declspec(dllexport) int main(int argc, char* argv[])
+#else
 int main(int argc, char* argv[])
+#endif
 {
     SetupEnvironment();
```

**Comment by ajtowns** [comment]
Rebased on top of #20715 and updated to use GetCommand, also removed some commented-out debug code.

File: contrib/signet/miner [hunk 1]
```diff
@@ -428,10 +428,13 @@ def do_generate(args):
             action_time = now
```
[the remark this answers is not in the document]
Comment (ajtowns) [reply contrib/signet/miner:428]: Gone

**Comment by MarcoFalke** [comment]
Forgot about this because it wasn't tagged 22.0

**Comment by laanwj** [comment]
Code review ACK b3c712cb2871e0034898b37454fb6a4e1fc78f84
(and tested the help functionality)

**Comment by kallewoof** [comment]
Post-merge ACK

**Pull request merged**